impeller/renderer/render_pass.h

```diff
@@ -44,20 +46,37 @@ class RenderPass {
   /// Note: this is not the native command buffer.
   void ReserveCommands(size_t command_count) {
     commands_.reserve(command_count);
+    bound_buffers_.reserve(command_count * 2 + 32);
+    bound_textures_.reserve(command_count);
   }
 
   HostBuffer& GetTransientsBuffer();
 
   //----------------------------------------------------------------------------
   /// @brief      Record a command for subsequent encoding to the underlying
-  ///             command buffer. No work is encoded into the command buffer at
-  ///             this time.
+  ///             command buffer, along with its binding information. No work is
+  ///             encoded into the command buffer at this time.
   ///
   /// @param[in]  command  The command
   ///
   /// @return     If the command was valid for subsequent commitment.
   ///
-  bool AddCommand(Command&& command);
+  bool AddCommand(Command&& command,
+                  std::initializer_list<BoundBuffer> buffers = {},
+                  std::initializer_list<BoundTexture> textures = {});
+
+  //----------------------------------------------------------------------------
+  /// @brief      Record a command for subsequent encoding to the underlying
+  ///             command buffer, along with its binding information. No work is
+  ///             encoded into the command buffer at this time.
+  ///
+  /// @param[in]  command  The command
+  ///
+  /// @return     If the command was valid for subsequent commitment.
+  ///
+  bool AddCommand(Command&& command,
+                  std::vector<BoundBuffer> buffers,
+                  std::vector<BoundTexture> textures);
```

Comment:
Why do we have this option?  There is a lot of duplicated code here between these implementations.

Reply:
Ahh this one is a good question for you 😄 

For some of the contents (like runtime effect) we do need to manage a vector allocation because the binding is very dynamic. I had thought that if I only had the method that accepted vectors, then the initializer lists would always be converted to vectors first, whereas I'm trying to avoid extra heap allocations.

Reply:
If we don't know the size at compile-time we can't avoid the heap allocation (safely).  If you make the method take in iterators you can share the same implementation and have the std::initializer_list and std::vector versions just call into that for convenience.

Reply:
Right, i'm not worried about avoiding that for runtime effect - but for things like solid color contents I am worried.

I like the idea of using iterators, I will try that out.